public/app/plugins/datasource/grafana/components/QueryEditor.tsx

```diff
@@ -367,6 +424,28 @@ export class QueryEditor extends PureComponent<Props, State> {
     onRunQuery();
   };
 
+  onFileDrop = (result: ArrayBuffer | String | null) => {
+    const snapshot: DataFrameJSON[] = [];
+
+    if (result) {
+      if (!result || result instanceof String) {
+        return;
+      }
+      const dataFrames = readSpreadsheet(result);
+      dataFrames.forEach((df) => {
+        const dataframeJson = dataFrameToJSON(df);
+        snapshot.push(dataframeJson);
+      });
+    }
+
+    this.props.onChange({
+      ...this.props.query,
+      queryType: GrafanaQueryType.Snapshot,
+      snapshot,
```

Comment:
is the uploaded csv stored in the query model (and hence dashboard model)? That seems like a potential problem

Reply:
Yes, the dataframes parsed from the spreadsheets are stored in the query model, and yes this could be a problem. We've kept the maximum allowed file size low to see how this works out. Eventually the data should move to the Object store.

Reply:
I was thinking about the same thing but sounds like you have a plan to manage this moving forward 🙌🏻